### PR TITLE
#1862 insurance selection

### LINKED ISSUE #1862: Insurance selection dropdown
## Is your feature request related to a problem? Please describe.

Need a insurance selector drop down for payments as well as discounting the total

## Describe the solution you'd like

Add a dropdown to select a insurance policy, and apply that policies discount to the total of the prescription

## Implementation

N/A

## Describe alternatives you've considered

N/A

## Additional context

N/A

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #1862 

## Change summary

- Add a insurance selection drop down. This alters the total based on insurance discount.

image

image

## Testing

N/A

### Related areas to think about

This is using a native drop down rather than the rubbish I wrote. I think this is a far better option as it is more fluid and efficient and already handles a few cases that I hadn't implemented yet - i.e. scrolling.

### CODE REVIEW

File: src/selectors/payment.js [hunk 1]
```diff
@@ -13,11 +13,11 @@ export const selectPrescriptionSubTotal = ({ payment }) => {
 };
 
 export const selectPrescriptionTotal = ({ payment }) => {
-  const { insurancePolicy } = payment;
-
+  const { policy } = payment;
+  console.log(!!policy);
```
Comment (mark-prins) [inline src/selectors/payment.js]: test code?

File: src/widgets/PaymentSummary.js [hunk 2]
```diff
@@ -58,36 +63,59 @@ const PaymentSummaryComponent = ({
...
+  const policyNumbers = React.useMemo(() => insurancePolicies.map(p => p.policyNumber), [
+    insurancePolicies,
+  ]);
+  const onSelectPolicy = React.useCallback((_, index) => choosePolicy(insurancePolicies[index]), [
+    choosePolicy,
+  ]);
+  return (
+    <ScrollView>
+      <FlexView flex={1} style={localStyles.container}>
+        <Text style={localStyles.title}>Payment</Text>
```
Comment (mark-prins) [inline src/widgets/PaymentSummary.js]: another one for the translation pile

File: src/widgets/PaymentSummary.js [hunk 3]
```diff
@@ -58,36 +63,59 @@ const PaymentSummaryComponent = ({
...
+
+          <FlexView flex={0.25}>
+            <NumberLabelRow
+              size="small"
+              text="Credit used"
+              isCurrency
+              number={creditUsed.format()}
+            />
+            <Text style={localStyles.errorMessageStyle}>
+              {creditOverflow ? 'Not enough credit!' : ''}
```
Comment (mark-prins) [inline src/widgets/PaymentSummary.js]: I thought I saw this in another PR? probably mistaken.

File: src/widgets/PaymentSummary.js [hunk 4]
```diff
@@ -97,6 +125,9 @@ PaymentSummaryComponent.propTypes = {
   paymentAmount: PropTypes.object.isRequired,
   creditOverflow: PropTypes.bool.isRequired,
   isComplete: PropTypes.bool.isRequired,
+  insurancePolicies: PropTypes.object.isRequired,
+  choosePolicy: PropTypes.func.isRequired,
```
Comment (mark-prins) [inline src/widgets/PaymentSummary.js]: can't see the assignment/action of `choosePolicy`?

File: src/selectors/payment.js (hunk 1 above)
Comment (joshxg) [reply src/selectors/payment.js]: :(


65f48f9

**Pull request merged**